### ISSUE TITLE
Full Text Index Quick Search Error

### ISSUE THREAD

**Issue by MaradiagaSistemas** [volkanceylan/Serenity#2788]
Hello,

Thanks for this amazing framework, I Recently add QuickSearch Attribute with FullTextContains type, but it throws an error while typing words with whitespaces between.

Thanks for your time and support.

![error](https://user-images.githubusercontent.com/1533653/31785624-fa93a692-b4c2-11e7-9276-746b49a0d29e.jpg)

**Comment by JohnRanger** [comment]
Hi MaradiagaSistemas,

Looks for me like a SQL error because your "condition" is called "Maradiaga Figueroa" which has a space between the two words.

See this SQL 2008 Stack Overflow question: [https://stackoverflow.com/questions/9435119/syntax-error-near-of-in-the-full-text-search-condition-control-of](url)

I would recommend to enclose the two words with double quotes in the search box.

With Kind regards,

John

**Comment by volkanceylan** [comment]
Yes, i probably should, marcobisio also wants to convert quick filter to some attribute which we can also use in advanced filter, so fix pending

**Comment by volkanceylan** [comment]
@JohnRanger, @marcobisio i've looked at CONTAINS syntax and it seems like putting search term in double quotes if it contains a space is not always right idea.

E.g. 'One AND Two' or 'One OR Two' is perfectly valid search terms. If i auto quote, i'll be preventing using similar queries. What do you think?

**Comment by marcobisio** [comment]
Hi @volkanceylan and @JohnRanger,
sorry for the late response, but int the last weeks I have been very busy!
I agree with you: putting search term in double quotes is not always a good idea.
I'm still on an older version of Serenity core with my custom FullTextIndex implementation, so I can't test the last version on a real scenario, but what I think is that we should support the [boolean operators.](https://docs.microsoft.com/en-us/sql/relational-databases/search/query-with-full-text-search#Using_Boolean_Operators).
So, for example, the search term **Volkan Ceylan OR Marco Bisio** should become **"Volkan Ceylan" OR "Marco Bisio"** because both the left part and the right part contains spaces.
So basically we need to parse the search term and look for the allowed boolean operators and then use double quotes only when a part of the search term contains spaces.
The other way round,  **Volkan Ceylan OR Marco** should become **"Volkan Ceylan" OR Marco** because the right part is a single word without spaces.

Now, I know, this is a very simplistic approach because boolean operatos are common words, but normally they are already considered as "noise words" so they don't impact so much the search results (and the relative rankings).

Let me know if you have a better idea/approach and obviously I'm available to do some test in my environment.

Thanks
Bye

**Comment by volkanceylan** [comment]
Hi @marcobisio, thanks for comments, i don't think we should implement Full Text Search parsing into Serenity, trying to implement such things generally fails in several cases, and may not be compatible with other DBs.

**Comment by JohnRanger** [comment]
HI @volkanceylan  and @marcobisio ,

sorry for the late Reply.

I have two proposals here:

**(1)**
What about the following simple automatic Approach:

1st: Just send the Content of the quick search down to SQL. 
2nd: if it causes a SQL error, then Quote the whole Thing and send it down again to SQL.

With this Approach, the "Standard" user (who does not know about the advanced SQL search) can find Things naturally - even with spacees inside --- and the advanced user can do his/her advanced query.

**Problem:**
With this approach, I see the Problem in the fact that starting to type and wait only a fraction of a second already fires the search - so advanced queries will nearly always be incomplete when fired first against the SQL backend and causing so SQL Errors.

**(2)**
Why not providing an additional button (like the "X" within the quick search field which would allow a user to Switch to advanced query type (e.g. opening a popup with a data entry line and an explicit "Search" button to initiate the query?)

--> This would allow to always Quote the standard quick search Input field data (with auto-query after the short timeout) and also allow advanced users to do the sofisticated queries without "haste" (and quoting as they want within their query).

What do you think?

With Kind regards,

John

**Comment by AlperanQ** [comment]
Trial and error is something we should avoid. We will later look at this again. Closing for now.

**Comment by marcobisio** [comment]
This isn't "trial and error" approach, but an open discussion on how to include this useful feature in Serenity.
